### ISSUE TITLE
Pending tasks not implemented for AMQP, and for redis it returns wrong value

### ISSUE THREAD

**Issue by bseenu** [RichardKnop/machinery#256]
Get pending tasks is not implemented for AMQP broker, and for redis it always returns 11 because of the code - https://github.com/RichardKnop/machinery/blob/master/v1/brokers/redis.go#L193

**Comment by RichardKnop** [comment]
@bseenu For AMQP it is not implemented as there are technical limitations, RabbitMQ is a good message queue but is not appropriate to use for storing data, therefor the AMQP backend is just a hack which can be used for development or testing but for any real usage I recommend to use AMQP broker and Memcache/Redis backend.

For the Redis bug, you can submit a PR to fix it if it's something obvious.

**Comment by bseenu** [comment]
Richard,

Thanks for responding, I am checking the pending message count in the queue

`	messages_url := `http://localhost:15672/api/queues/%2f/machinery_task`
	req, _ := http.NewRequest("GET", messages_url, nil)
       client := &http.Client{}
       http_resp, _ := client.Do(req)
		defer http_resp.Body.Close()
		var data map[string]interface{}
		rbytes, _ := ioutil.ReadAll(http_resp.Body)
		json.Unmarshal(rbytes, &data)
		if data["messages"] == float64(0) {
                 fmt.Printf("No more pending tasks")
             } else {
                fmt.Printf("Pending tasks %g\n, data["messages"]
           }`

Though this will not take care of the cases where messages are being re-tried as they go to seperate queue, but is it possible to check for all the possible queues to determine the outstanding tasks

**Comment by bseenu** [comment]
Also any reason why you want to put the retry in a seperate queue instead of same one?

**Comment by RichardKnop** [comment]
@bseenu Right, I guess you can do that via HTTP API, not sure what information you get back. Do you get back whole message payload? Because I'm not using HTTP, just communicating with RabbitMQ via the AMQP library and there was no way to reading messages from a queue without consuming them (e.g. taking them off the queue) which is why this functionality is not implemented for AMQP.

If you think this can work well via the HTTP API, you can make a pull request :)

Also for the Redis bug which seems like a simple fix :)

**Comment by Blogoslov** [comment]
For Redis bug is fixed. I see that pending tasks are correct now.
